src/ConfigCatProvider.tsx

```diff
@@ -1,31 +1,86 @@
 import React, { Component, PropsWithChildren } from "react";
+import * as configcatcommon from "configcat-common";
+import { PollingMode } from "./PollingMode";
+import { HttpConfigFetcher } from "./ConfigFetcher";
+import { LocalStorageCache } from "./Cache";
+import CONFIGCAT_SDK_VERSION from "./Version";
 import ConfigCatContext from "./ConfigCatContext";
-import { IConfigCatClient } from "configcat-common";
+import { IReactAutoPollOptions, IReactLazyLoadingOptions, IReactManualPollOptions } from "./Exports";
 
 type ConfigCatProviderProps = {
-  client: IConfigCatClient;
+  sdkKey: string;
+  pollingMode?: PollingMode | undefined;
+  options?: IReactAutoPollOptions | IReactLazyLoadingOptions | IReactManualPollOptions | undefined;
 };
 
 type ConfigCatProviderState = {
-  client: IConfigCatClient;
+  client: configcatcommon.IConfigCatClient;
 };
 
 class ConfigCatProvider extends Component<
   PropsWithChildren<ConfigCatProviderProps>,
   ConfigCatProviderState,
   {}
 > {
+
   constructor(props: ConfigCatProviderProps) {
     super(props);
+    const client = this.initializeConfigCatClient();
+    this.state = { client };
+  }
+
+  componentDidUpdate() {
+    this.state?.client?.dispose();
+    const client = this.initializeConfigCatClient();
+    this.setState({ client });
+  }
 
-    // TODO remove when client supports 'ready' callback
-    this.state = { client: props.client };
+  componentWillUnmount() {
+    this.state?.client?.dispose();
   }
 
-  componentDidUpdate(props: ConfigCatProviderProps) {
-    this.setState({
-      client: props.client,
-    });
+  private initializeConfigCatClient() {
+    let client: configcatcommon.IConfigCatClient;
+    switch (this.props.pollingMode) {
+      case PollingMode.LazyLoad:
+        client = configcatcommon.createClientWithLazyLoad(
+          this.props.sdkKey,
+          {
+            configFetcher: new HttpConfigFetcher(),
```

Comment:
It would be nice if you extracted the option into the const variable and reused it in all places.

Reply:
done